### ISSUE TITLE
Checkout on PRs sometimes fail - couldn't find remote ref refs/pull/XX/merge

### ISSUE THREAD

**Issue by dontrolle** [mangrovedao/mangrove-ts#497]
**Describe the bug**
Sometimes the checkout step of our GitHub Actions Workflows fails, with the following fault:

` Error: The process '/usr/bin/git' failed with exit code 128`

Using this PR, https://github.com/mangrovedao/mangrove-ts/runs/6692239678?check_suite_focus=true, as an example the core error is that the `git` `fetch` step:

`/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +refs/pull/488/merge:refs/remotes/pull/488/merge` 

fails with a 

`   Error: fatal: couldn't find remote ref refs/pull/488/merge`

The `refs/pull/[no]/merge` refspec is a `[standard](https://git-scm.com/book/id/v2/GitHub-Maintaining-a-Project)` way of referring to the commit that would result, should the PR be merged. Consequentially, GitHub should always respond with something reasonable when fetch on this refspec. 

**To Reproduce**
Can't reproduce consistently, as it relies on errors on GitHub-side.

**To sum up**

Our conclusion is that the problem lies with transient errors on GitHubs side, not (for instance) with the https://github.com/actions/checkout (however worryingly many issues that most used action on GitHub has).

We log this bug to allow us to track and comment on errors related to this problem, but will mark it blocked for now. Initial inquiries to GitHub support for help resolving this have proven rather unhelpful.

┆Issue is synchronized with this [Monday item](https://mangrovedao.monday.com/boards/3894180513/pulses/3902013859) by [Unito](https://www.unito.io)

**Comment by peterMangrove** [comment]
@dontrolle Is this still an issue?

**Comment by dontrolle** [comment]
If we haven't seen it lately, then let's close it. Closing.
